src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +194,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (isset($request['name'])) {
+                $options['name'] = $request['name'];
+            }
+
+            if (isset($request['attached_files'])) {
```

Comment:
like the `options` key, I'll always set the `attached_files` key in `add()`, and then check here for `!empty($request['attached_files'])`